pypeit/edgetrace.py

```diff
@@ -0,0 +1,4122 @@
+
+    def load(self, validate=True, rebuild_pca=True):
+        """
+        Load and reinitialize the trace data.
+
+        Data is read from :attr:`file_path` and used to overwrite any
+        internal data. Specific comparisons of the saved data are
+        performed to ensure the file is consistent with having been
+        written by a consistent version of the code; see
+        :func:`_reinit`.
+
+        To load a full :class:`EdgeTraceSet` directly from a file,
+        use :func:`from_file`.
+
+        Args:
+            validate (:obj:`bool`, optional):
+                Validate that the spectrograph, parameter set, and
+                bitmask have not changed between the current internal
+                values and the values read from the fits file. The
+                method raises an error if the spectrograph or
+                parameter set are different. If the bitmask is
+                different, a warning is issued and the bitmask
+                defined by the header is used instead of the existing
+                :attr:`bitmask`.
+            rebuild_pca (:obj:`bool`, optional):
+                If the primary header indicates that the PCA
+                decompostion had been performed on the save object,
+                use the saved parameters to rebuild that PCA.
+        Raises:
+            FileNotFoundError:
+                Raised if no data has been written for this master
+                frame.
+            ValueError:
+                Raised if validation of the data fails (actually
+                raised by :func:`_reinit`).
+        """
+        # Check the file exists
+        if not self.exists():
+            msgs.error('File does not exit: {0}'.format(self.file_path))
+        with fits.open(self.file_path) as hdu:
+            # Re-initialize and validate
+            self._reinit(hdu, validate=validate, rebuild_pca=rebuild_pca)
+
+    def exists(self):
+        """
+        Check if the output file already exists.
+        """
+        return os.path.isfile(self.file_path)
+
+    @classmethod
+    def from_file(cls, filename, rebuild_pca=True):
+        """
+        Instantiate using data from a file.
+
+        To reload data that has been saved for an existing
+        instantiation, use :func:`load`.
+
+        Args:
+            filename (:obj:`str`):
+                Fits file produced by :func:`save`.
+            rebuild_pca (:obj:`bool`, optional):
+                If the primary header indicates that the PCA
+                decompostion had been performed on the save object,
+                use the saved parameters to rebuild that PCA.
+        """
+        # Check the file exists
+        if not os.path.isfile(filename):
+            msgs.error('File does not exit: {0}'.format(filename))
+        msgs.info('Loading EdgeTraceSet data from: {0}'.format(filename))
+        with fits.open(filename) as hdu:
+            this = cls(load_spectrograph(hdu[0].header['PYP_SPEC']),
```

Comment:
You might use my somewhat hacky `items_from_master_file()` method
in masterframe.py to "standardize" things.  But I'll forgive you if you don't.

Reply:
I need to pass the whole hdu to `_reinit`, so I'd rather keep it as is, instead of changing `items_from_master_file`.  But we should probably come back to this.  Added a comment.